src/contact.rs

```diff
@@ -1603,7 +1603,7 @@ pub(crate) async fn set_profile_image(
     if changed {
         contact.update_param(context).await?;
         context.emit_event(EventType::ContactsChanged(Some(contact_id)));
-        ui_events::emit_chatlist_item_changed_for_contacts_dm_chat(context, contact_id);
+        ui_events::emit_chatlist_item_changed_for_contact_chat(context, contact_id).await;
```

Comment:
Why is it not sufficient for the app to know the contact ID? We better add `contact_id` + `chat_type` (1:1, group, etc.) to `struct Summary` so that the app can iterate over summaries and quickly find the chatlist item changed w/o doing extra db queries.

Reply:
this is about the dm chat with that contact (profile image and recently online), the db query for that should be cheap and saves us a round trip to core. But sure we can also discuss creating new dedicated events for it, my idea here is to have some small amount of chatlist events that the ui listens for. (with the motto: if you listen for all events with chatlist in the name you are set)

Reply:
Yes, this db query is cheap, but if we just add all necessary info to `struct Summary`, we don't need any additional events at all. So, having `EventType::ContactsChanged` is sufficient. Also we don't need an extra round trip to the core because, i guess, the app already has the list of all summaries (it should be built to display the chat list) and can iterate over them (actually only over currently visible ones) to find out which items has changed.

Reply:
Having all these different events to listen to is a mess, my goal here is to move the concern to core to give out events when the (granted virtual) resource of chatlist changes (be it order or item content).

I also plan to add tests for all the cases to core.

So in my mind we could also do this (I don't want to discuss exact naming at this step):
- for avatar and recently online changes
  - add contact id to chat list items of DM chats
  - add `ChatlistContactDirectMessagesChatChanged`
- add `ChatlistContactNameChanged`

I am against reusing `EventType::ContactsChanged` as those events are not precise enough and you have many of them to handle, so a new feature comes around and it needs a new event to be listened to in the UI, the point of this pr/effort is to make the ui part much easier that you exactly know which few events you need to handle to have handled all cases for when the chatlist updates, ideally as precise as (reasonable in db time constrains) possible, that the lazy virtual list in desktop can be partially updated as needed and not reload all chats when one of them changes.